### DISCUSSION TITLE
Pipeline file access within code

### DISCUSSION THREAD

**Discussion by crooktec-tannercrook** [dataplane-app/dataplane#563] [Q&A] [answered]
When editing code in the pipeline there is a place where you can create files. Is there anyway to reference these files in the python script? I have a package that requires a json keyfile and it is the one thing that is blocking me from building that pipeline.

**Comment by saul-data** [comment, marked as answer]
Hi @crooktec-tannercrook, if the json file is small, you could save the json file in code editor by simply creating a file. If it is a large json file, I'd recommend storing that file in a database like Postgresql or S3 and then pulling it from that storage. 

Here is an example:
<img width="506" alt="image" src="https://user-images.githubusercontent.com/63714857/217584205-f455cb48-efa9-459f-ac5f-1a04d628e49f.png">

<img width="748" alt="image" src="https://user-images.githubusercontent.com/63714857/217584428-817cb5e7-2fe8-40d5-ab2f-9931caa86341.png">

You should then reference that file using a relative path in your python code. 
<img width="1385" alt="image" src="https://user-images.githubusercontent.com/63714857/217586713-ee4b81a6-b27c-4609-95c1-01490271415d.png">

```python
import json

# Get scripts directory
import pathlib
jsonpath = pathlib.Path(__file__).parent.resolve()
print(jsonpath)
  
# Opening JSON file
f = open(str(jsonpath)+'/myjsonfile.json')
  
# returns JSON object as 
# a dictionary
data = json.load(f)

print(data)
```

**Reply by crooktec-tannercrook** [reply]
This is exactly what I needed. Thank you! I will try to put some documentation together so I can help contribute as I think this is gold when migrating from an airflow solution.

**Reply by saul-data** [reply]
Thank you. 😊 I’m glad this helped you. Agreed we need more recipes and documentation, any assistance with contribution will be greatly appreciated by us and the data community. There are some exciting new features coming in soon including a package manager so you can build your own Python packages to share code or functions across the pipelines.
